lib/Backend/GlobOpt.cpp

```diff
@@ -13525,16 +13526,19 @@ GlobOpt::OptStackArgLenAndConst(IR::Instr* instr, Value** src1Val)
                 {
                     int argIndex = indirOpndSrc1->GetOffset() + 1;
                     IR::Instr* defInstr = nullptr;
-                    IR::Instr* inlineeStart = instr->m_func->GetInlineeStart();
-                    inlineeStart->IterateArgInstrs([&](IR::Instr* argInstr) {
-                        StackSym *argSym = argInstr->GetDst()->AsSymOpnd()->m_sym->AsStackSym();
-                        if (argSym->GetArgSlotNum() - 1 == argIndex)
-                        {
-                            defInstr = argInstr;
-                            return true;
-                        }
-                        return false;
-                    });
+                    if (argIndex > 0)
+                    {
+                        IR::Instr* inlineeStart = instr->m_func->GetInlineeStart();
+                        inlineeStart->IterateArgInstrs([&](IR::Instr* argInstr) {
+                            StackSym *argSym = argInstr->GetDst()->AsSymOpnd()->m_sym->AsStackSym();
+                            if (argSym->GetArgSlotNum() - 1 == argIndex)
+                            {
+                                defInstr = argInstr;
+                                return true;
+                            }
+                            return false;
+                        });
+                    }
```

Comment:
This is the bugFix, in the case that `arguments[-1]` was used, argIndex would be 0 (see line 13527) but there would be an argSym with slot value 0 - namely the `this` object. Avoid incorrectly matching it by not searching for a match if the index is negative.

`arguments[-n]` for any `n` should return `undefined` - which is what we produce below if `defInstr` is left as `nullptr`

Reply:
So you say this is assuming that arguments[-n] is always "undefined"? That doesn't sound right.

+function() {
    arguments[-1] = "it's me!";
    WScript.Echo(arguments[-1]);
}();

Reply:
A very good point, I need to look at how this case is handled - and add a test for it. I've converted the PR to draft for now as this may make it significantly more complicated.

Reply:
I think this should be fairly simple. As with arrays, a load of a[n] where n is not in 0...a.length-1 should just be forwarded to the load-element runtime helper. That case isn't important to optimize anyway, and the semantics around array gaps, prototype objects, etc., aren't something we should try to represent in jitted code.

Reply:
BTW, it's weird to me that this actually passed the unit tests. We're obviously missing some coverage for this kind of case.

Reply:
There is certainly missing coverage.

But also currently the optimisation is disabled in advance if the arguments object is written to - so for any out of range numerically indexed property field the correct value within the optimisation is undefined.

I think the only currently detectable incorrect behaviour is with arguments[-1] specifically in an inlined function picking up the this value.

My attempt to remove a few rejits may add another incorrect case though I'd still like to double check that.